projectmanagement/src/main/java/life/qbic/projectmanagement/offer/OfferId.java

```diff
@@ -0,0 +1,39 @@
+package life.qbic.projectmanagement.offer;
+
+import static java.util.Objects.requireNonNull;
+
+import life.qbic.projectmanagement.ProjectManagementDomainException;
+
+/**
+ * The id of an offer.
+ *
+ * @since <version tag>
+ */
+public record OfferId(ProjectConservedPart projectConservedPart, RandomPart randomPart,
+                      Version version) {
+
+  public OfferId {
+    requireNonNull(projectConservedPart);
+    requireNonNull(randomPart);
+    requireNonNull(version);
+  }
+
+  public static OfferId from(String offerId) {
+    //O_[project conserved part]_[random id]_[version]
+    if (!offerId.startsWith("O_")) {
+      throw new ProjectManagementDomainException(
+          "OfferId does not start with 'O_'. Provided: " + offerId);
+    }
+    String[] splits = offerId.split("_");
+    if (splits.length != 4) {
+      throw new ProjectManagementDomainException(
+          "OfferId does not contain 4 parts. Provided: " + offerId);
+    }
+    ProjectConservedPart projectConservedPart = new ProjectConservedPart(splits[1]);
+    RandomPart randomPart = new RandomPart(splits[2]);
+    long versionNumber = Long.parseLong(splits[3]);
+    Version version = new Version(versionNumber);
+    return new OfferId(projectConservedPart, randomPart, version);
```

Comment:
This is business logic from the finance and accounting domain and should not reside in the project management domain.

I suggest to use a simple internal character sequence representation of the offer id, until we have a reason to change it.

Reply:
Yes this domain did not exist when I implemented these classes. That is why I implemented them :D